src/catalog/trigger_catalog.cpp

```diff
@@ -0,0 +1,302 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// query_metrics_catalog.cpp
+//
+// Identification: src/catalog/query_metrics_catalog.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "catalog/catalog.h"
+#include "catalog/trigger_catalog.h"
+#include "catalog/catalog_defaults.h"
+#include "common/macros.h"
+#include "common/logger.h"
+
+namespace peloton {
+namespace catalog {
+
+TriggerCatalog *TriggerCatalog::GetInstance(
+    concurrency::Transaction *txn) {
+  static std::unique_ptr<TriggerCatalog> trigger_catalog(
+      new TriggerCatalog(txn));
+
+  return trigger_catalog.get();
+}
+
+TriggerCatalog::TriggerCatalog(concurrency::Transaction *txn)
+    : AbstractCatalog("CREATE TABLE " CATALOG_DATABASE_NAME
+                      "." TRIGGER_CATALOG_NAME
+                      " ("
+                      "trigger_oid           INT NOT NULL, "
+                      "trigger_name          VARCHAR NOT NULL, "
+                      "database_oid          INT NOT NULL, "
+                      "table_oid             INT NOT NULL, "
+                      "trigger_type          INT NOT NULL, "
+                      "fire_condition        VARCHAR, "
+                      "function_name         VARCHAR, "
+                      "function_arguments    VARCHAR, "
+                      "time_stamp            INT NOT NULL);",
+                      txn) {
+  // Add secondary index here if necessary
+  Catalog::GetInstance()->CreateIndex(
+      CATALOG_DATABASE_NAME, TRIGGER_CATALOG_NAME,
+      {"database_oid", "table_oid", "trigger_type"}, TRIGGER_CATALOG_NAME "_skey0",
+      false, IndexType::BWTREE, txn);
```

Comment:
Good point. But I find it is needed to specify the type of index here. FYI, there is a comment about the type of catalog index in `catalog.cpp`: `This should be hash index rather than tree index?? (but postgres use btree!!)`